runtime/src/qio/qio_formatted.c

```diff
@@ -361,6 +361,84 @@ qioerr _append_char(char* restrict * restrict buf, size_t* restrict buf_len, siz
   return 0;
 }
 
+qioerr read_entire_file(const int threadsafe,qio_channel_t* restrict ch, const char* restrict* restrict out, int64_t* restrict len_out, ssize_t maxlen)
+{  
+  qioerr err,err_reading_file=0;
+  int64_t file_length;
+  char* ret=NULL;
+  ssize_t amt=0;
+  err_t errcode;
+  int64_t fileLenGuess=1024;
+  ssize_t total_amt_read=0;
+  bool err_flag;
+  if( threadsafe ) 
+  {
+    err = qio_lock(&ch->lock);
+    if( err ) return err;
+  }
+
+  err=qio_file_length(ch->file,&file_length);   // obtain file length
+ 
+  if(!err)                                      // no error in obtaining file length
+  {
+   
+    ret=qio_malloc(file_length+1);              //allocate initial buffer
+    ret[0]='\0';
+    err=qio_channel_read(false, ch, ret,file_length+1, &amt);    //file_length+1 instead of file_length -->I try to read one more character than the prescribed 
```

Comment:
We keep the source code to 80 column lines. Could you move the comments to be before the lines that they apply to?